packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterNativeAd.java

```diff
@@ -139,9 +141,17 @@ void load() {
       @Override
       public void onNativeAdLoaded(@NonNull NativeAd nativeAd) {
         ad = adFactory.createNativeAd(nativeAd, customOptions);
+        responseInfo = nativeAd.getResponseInfo();
       }
     };
-    AdListener adListener = new FlutterAdListener(manager, this) {
+
+    ResponseInfoProvider responseInfoProvider = new ResponseInfoProvider() {
```

Comment:
```suggestion
    final ResponseInfoProvider responseInfoProvider = new ResponseInfoProvider() {
```

just for consistency. The style guide doesn't say that `final` must be used, so if we want to change the need for it, it's probably ok.